### PR TITLE
In payment 32

### PR DESCRIPTION
**Pull request by lipob**
### What this PR does (required):
- The user can create a new payment method or update an existing one. 
- If the user isn't a customer yet, first it creates a customer. 
- Then attach the payment method to the customer and set the default payment method.

### Screenshots / Videos (required):
Payment method visa
Payment method Mastercad

### CODE REVIEW

File: client/src/helpers/APICalls/createCustomer.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Customer } from '../../interface/ProfileApiData';
+import { FetchOptions } from '../../interface/FetchOptions';
+
+const createCustomer = async (id: string): Promise<Customer> => {
+  const fetchOptions: FetchOptions = {
+    method: 'POST',
+    credentials: 'include',
+  };
+  console.log(id);
```
Comment (ezeilo-su) [inline client/src/helpers/APICalls/createCustomer.ts:9]: Remove `console.log`

File: server/controllers/payment.js [hunk 2]
```diff
@@ -3,6 +3,71 @@ const Profile = require('../models/Profile');
...
 
+exports.createCustomer = asyncHandler(async (req, res) => {
+  const user = await Profile.findById(req.params.id);
+  const { email } = user;
+
+  try {
+    const customer = await stripe.customers.create({email});
+    
+    // update user in DB
+    user.customerId = customer.id;
```
Comment (ezeilo-su) [inline server/controllers/payment.js:14]: Make sure that user exists. You can validate user first before the update

File: server/controllers/payment.js [hunk 3]
```diff
@@ -3,6 +3,71 @@ const Profile = require('../models/Profile');
...
+  }
+  catch (err) {
+    res.status(400);
+    throw new Error(err, "Error creating customer");
+  }
+});
+
+exports.retrieveCustomer = asyncHandler(async (req, res) => {
+  try {
+    const customer = await stripe.customers.retrieve(req.params.id);
```
Comment (ezeilo-su) [inline server/controllers/payment.js:27]: First validate customer

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Great work on this one. Those screenshots are nice. Check out a few comments left below so you can address them.

File: client/src/components/Payment/Payment.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  });
+
+  // Fetch user's profile.
+  useEffect(() => {
+    setLoading(true);
+    const fetchUser = async () => {
+      if (loggedInUser) {
+        const data: CustomerProfile = await getCustomerProfile(loggedInUser.id);
+        setUserProfile(data);
+        if (data.customerId.length) {
```
[the remark this answers is not in the document]
Comment (lipob) [reply client/src/components/Payment/Payment.tsx:58]: Hi Sunday. I'm validating if there is a username for this user profile if it isn't an empty string. How do I have to validate it?

File: client/src/App.tsx [hunk 5]
```diff
@@ -20,6 +23,10 @@ import './App.css';
 import NavBar from './components/NavBar/NavBar';
 import AllNotifications from './pages/AllNotifiactions/AllNotifications';
 
+const stripePromise = loadStripe(
+  'pk_test_51JYBH4HXzYVsCZt0Mls8k6b8UIXWlzgUTCinfBuNzmrqYLlUeUQuV4gcj7sePE1cDlUP2sRkFWJmAMgqovjZOEMd006aTSQqIg',
```
Comment (moffatethan) [inline client/src/App.tsx:27]: This would better belong in an environment variable. https://create-react-app.dev/docs/adding-custom-environment-variables/

**Comment by moffatethan** [comment]
@lipob Please fix the merge conflict

File: client/src/App.tsx (hunk 5 above)
Comment (lipob) [reply client/src/App.tsx]: Thanks, Ethan. I was looking for that docs

**Comment by lipob** [comment]
Conflicts fixed. Please let me know if I can merge this PR

**Review by moffatethan — APPROVED** [review]

**Pull request merged**